internal/core/textlayout.rs

```diff
@@ -532,6 +537,26 @@ fn test_cursor_position_with_newline() {
     assert_eq!(paragraph.cursor_pos_for_byte_offset(5), (5. * 10., 0.));
 }
 
+#[test]
+fn byte_offset_for_empty_line() {
+    let font = FixedTestFont;
+    let text = "Hello\n\nWorld";
+
+    let paragraph = TextParagraphLayout {
+        string: text,
+        layout: TextLayout { font: &font, letter_spacing: None },
+        max_width: 100. * 10.,
+        max_height: 10.,
+        horizontal_alignment: TextHorizontalAlignment::Left,
+        vertical_alignment: TextVerticalAlignment::Top,
+        wrap: TextWrap::WordWrap,
+        overflow: TextOverflow::Clip,
+        single_line: false,
+    };
+
+    assert_eq!(paragraph.byte_offset_for_position((0., 10.)), 6);
+}
+
 #[test]
 fn test_byte_offset() {
```

Comment:
Please extend this test to verify that `paragraph.byte_offset_for_position` returns the last byte offset when the y position is below the last line. I believe that part isn't covered yet, or did I miss that?